Types/Request.php

```diff
@@ -44,28 +44,48 @@ class Request {
     /**
      * @var array
      */
-    private $curlOptions;
+    private $curlOptions = [];
 
     /**
      * @var string
      */
     private $query;
 
     /**
+     * @var int
+     */
+    private $expectedStatusCode = 200;
+
+    /**
      * Request constructor
      *
-     * @param string      $method
-     * @param string      $url
-     * @param array       $curlOptions
-     * @param string|null $query
-     * @param string|null $payload
-     */
-    public function __construct($method, $url, array $curlOptions, $query = null, $payload = null) {
-        $this->method      = $method;
-        $this->url         = $url;
-        $this->payload     = $payload;
-        $this->curlOptions = $curlOptions;
-        $this->query       = $query;
+     * @param array $options
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
+     */
+    public function __construct(array $options) {
+        HashMap::assert($options, 'options');
+        HashMapEntry::assertExists($options, 'method', 'options.method');
+        HashMapEntry::assertExists($options, 'url', 'options.url');
+
+        foreach ($options as $key => $value) $this->$key = $value;
```

Comment:
leerzeichen zwischen foreach un klammer auf (hasde sons anders gemacht in dem projekt)